pkg/app/pipedv1/plugin/kubernetes_multicluster/livestate/plugin.go

```diff
@@ -31,85 +31,148 @@ import (
 
 type Plugin struct{}
 
-// GetLivestate implements sdk.LivestatePlugin.
 func (p Plugin) GetLivestate(ctx context.Context, _ *sdk.ConfigNone, deployTargets []*sdk.DeployTarget[kubeconfig.KubernetesDeployTargetConfig], input *sdk.GetLivestateInput[kubeconfig.KubernetesApplicationSpec]) (*sdk.GetLivestateResponse, error) {
-	if len(deployTargets) != 1 {
-		return nil, fmt.Errorf("only 1 deploy target is allowed but got %d", len(deployTargets))
-	}
-
-	deployTarget := deployTargets[0]
-	deployTargetConfig := deployTarget.Config
-
 	cfg, err := input.Request.DeploymentSource.AppConfig()
 	if err != nil {
 		input.Logger.Error("Failed to load application config", zap.Error(err))
 		return nil, err
 	}
 
+	logger := input.Logger
+
+	type targetConfig struct {
+		deployTarget *sdk.DeployTarget[kubeconfig.KubernetesDeployTargetConfig]
+		multiTarget  *kubeconfig.KubernetesMultiTarget
+	}
+
+	deployTargetMap := make(map[string]*sdk.DeployTarget[kubeconfig.KubernetesDeployTargetConfig], 0)
+	targetConfigs := make([]targetConfig, 0, len(deployTargets))
+
+	// prevent the deployment when its deployTarget is not found in the piped config
+	for _, target := range deployTargets {
+		deployTargetMap[target.Name] = target
+	}
+
+	// If no multi-targets are specified, sync to all deploy targets.
+	if len(cfg.Spec.Input.MultiTargets) == 0 {
+		for _, dt := range deployTargets {
+			targetConfigs = append(targetConfigs, targetConfig{
+				deployTarget: dt,
+				multiTarget:  nil,
+			})
+		}
+	} else {
+		// Sync to the specified multi-targets.
+		for _, multiTarget := range cfg.Spec.Input.MultiTargets {
+			dt, ok := deployTargetMap[multiTarget.Target.Name]
+			if !ok {
+				logger.Info("Ignore multi target '%s': not matched any deployTarget", zap.String("multiTargetName", multiTarget.Target.Name))
+				continue
+			}
+
+			targetConfigs = append(targetConfigs, targetConfig{
+				deployTarget: dt,
+				multiTarget:  &multiTarget,
+			})
+		}
+	}
+
 	// TODO: find the way to hold the tool registry and loader in the plugin.
 	// Currently, we create them every time the stage is executed beucause we can't pass input.Client.toolRegistry to the plugin when starting the plugin.
 	toolRegistry := toolregistry.NewRegistry(input.Client.ToolRegistry())
 
-	// Get the kubectl tool path.
-	kubectlPath, err := toolRegistry.Kubectl(ctx, cmp.Or(cfg.Spec.Input.KubectlVersion, deployTargetConfig.KubectlVersion))
-	if err != nil {
-		input.Logger.Error("Failed to get kubectl tool", zap.Error(err))
-		return nil, err
+	liveStates := make([]sdk.ApplicationLiveState, 0, len(targetConfigs))
+	syncStates := make([]sdk.ApplicationSyncState, 0, len(targetConfigs))
+	for _, tc := range targetConfigs {
+		// Get the kubectl tool path.
+		kubectlPath, err := toolRegistry.Kubectl(ctx, cmp.Or(cfg.Spec.Input.KubectlVersion, tc.deployTarget.Config.KubectlVersion))
```

Comment:
This may be wrong. I think `tc.deployTarget.Config.KubectlVersion` has a higher priority.
```suggestion
		kubectlPath, err := toolRegistry.Kubectl(ctx, tc.deployTarget.Config.KubectlVersion, cmp.Or(cfg.Spec.Input.KubectlVersion))
```

Reply:
@Warashi 
It seems that `cfg.Spec.Input.KubectlVersion` has a higher priority in the old version.
 
> // priority: applicationConfig.KubectlVersion > pipedConfig.KubectlVersion
https://github.com/pipe-cd/pipecd/blob/f53dd3cacc5dab5db3df48332b98f36b53158f17/pkg/app/piped/platformprovider/kubernetes/applier.go#L209C1-L217C2


It is the same as other deployment loginc for k8s plugin and k8s_multicluster.

Reply:
@ffjlabo 
Oops, thank you. I am confused with deployTargets and multiTargets.
ref; https://github.com/pipe-cd/pipecd/blob/f53dd3cacc5dab5db3df48332b98f36b53158f17/pkg/app/pipedv1/plugin/kubernetes_multicluster/deployment/rollback.go#L182-L186